mmengine/hooks/checkpoint_hook.py

```diff
@@ -325,10 +331,13 @@ def after_train_epoch(self, runner) -> None:
         if not self.by_epoch:
             return
 
+        if runner.epoch + 1 < self.save_begin:
+            return
```

Comment:
If the logic of `every_n_epochs` is updated as I proposed, this check can be deleted.

Reply:
Hi @KerwinKai , this is the final comment. Thanks for your contribution.